src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -29,25 +41,122 @@ export default class FileField extends React.Component {
     }
   }
 
-  onAddFile = (event, index = null) => {
+  componentDidMount() {
+    this.setState({
+      // Check PDF password feature flag in a secondary connected component
+      // ref: https://medium.com/octopus-labs-london/how-to-access-a-redux-components-methods-with-createref-ca28a96efd59
+      pdfPasswordFeature: this.pdfPasswordFeature.current?.getWrappedInstance()
+        ?.props.requestLockedPdfPassword,
+    });
+  }
+
+  focusAddAnotherButton = () => {
+    // focus on span pseudo-button, not the label
+    focusElement(`#${this.props.idSchema.$id}_add_label span`);
+  };
+
+  /**
+   * @typedef Files~state - File state object
+   * @type {object}
+   * @property {object|null} file - upload API supplied File object: see
+   *   https://developer.mozilla.org/en-US/docs/Web/API/File
+   * @property {number} index - index matching formData array position
+   * @property {boolean} isEncrypted - detected encrypted state of a PDF file
+   * @property {boolean} isTemporary - indicates state data that needs needs a
+   *   placeholder
+   */
+  /**
+   * Update file state
+   * @param {object|null} file - API supplied File object; store the entire File
+   *   object because it needs to be passed to the `uploadRequest` function
+   * @param {number} index - index of file in formData
+   * @param {boolean} isEncrypted - encrypted state
+   */
+  setFileState = ({ file, index, isEncrypted }) => {
+    // remove already uploaded files from state
+    const files = this.state.files.filter(
+      fileObj =>
+        this.props.formData.findIndex(
+          fileData => fileData.name === fileObj.file.name,
+        ) < 0,
+    );
+
+    const existingIndex = files.findIndex(fileData => fileData.index === index);
+    const data = { file, index, isEncrypted, isTemporary: true };
+    if (existingIndex > -1) {
+      files[existingIndex] = data;
+      this.setState({ files });
+    } else {
+      this.setState({ files: [...files, data] });
+    }
+  };
+
+  onSubmitPassword = (file, index, password) => {
+    if (file && password) {
+      this.onAddFile({ target: { files: [file] } }, index, password);
+    }
+  };
+
+  processFile = async (file, index) =>
+    checkForEncryptedPdf(file)
+      .then(result => {
```

Comment:
Since `checkForEncryptedPdf`returns a boolean, can `result` be named to something like `isEncrypted`?